### DISCUSSION TITLE
PDF generated from local docker container is always empty (HTML/Chromium)

### DISCUSSION THREAD

**Discussion by koriner** [gotenberg/gotenberg#508] [Q&A] [answered]
Hi! 

I've been testing gotenberg to generate PDFs from a NextJS-generated HTML page. I set up the docker container on a free Render.com instance (as mentioned in the docs) - and this is working, PDFs are generated as expected.

But in order to test my stuff, I want to be able to run a local docker of gotenberg, and then point to another local server for the HTML.

e.g. Gotenberg on `localhost:3000` and my HTML page on `localhost:3001/something`. 

For some reason this is not working for me. When I try to generate them from my localhost server, the PDF is just blank and the file size is tiny.

Anybody know why this might be? I can work around it by using the one sitting on render.com, but that means I have to deploy all changes to my site before I can test the PDFs.

Thanks!

**Comment by gulien** [comment, marked as answer]
Hello @koriner,

On your Gotenberg container network, localhost means basically something running on `127.0.0.1` **in** your container;

That being said, with Docker Compose, you may easily create a shared network between two containers. For instance:

```yaml
version: "3"

services:

  your_container_exposing_html:
    image: ??

  gotenberg:
    image: gotenberg/gotenberg:7
```

Gotenberg will now be able to access to `your_containe_exposing_html:3001/something`

**Reply by koriner** [reply]
@gulien Thanks for such a quick reply. This makes perfect sense now (I'm new to Docker)!

**Reply by gulien** [reply]
No problem, have fun 😄
